alacritty_terminal/src/input.rs

```diff
@@ -935,21 +936,27 @@ impl<'a, A: ActionContext + 'a> Processor<'a, A> {
             };
 
             if is_triggered {
-                // binding was triggered; run the action
-                binding.execute(&mut self.ctx, false);
-                has_binding = true;
+                // Binding was triggered; run the action
+                Some(binding.execute(&mut self.ctx, false))
+            } else {
+                None
             }
-        }
-
-        has_binding
+        }).fold(None, |should_suppress_char, suppress_char| {
+            // Suppress char by default as soon as a binding has been triggered
+            let should_suppress_char = should_suppress_char.unwrap_or(true);
+            Some(should_suppress_char && suppress_char)
+        }).unwrap_or_else(|| {
+            // Do not suppress char if no bindings were triggered
+            false
+        })
```

Comment:
Generally I like this approach, but I'm a little sad to see `process_key_bindings` and `process_mouse_bindings` diverging. This might be an argument for continuing to have `execute` set `ctx.suppress_chars()` itself.

Thoughts?

Reply:
Can't have execute call that itself, otherwise binding order matters.